src/snmalloc/mem/corealloc.h

```diff
@@ -792,11 +796,16 @@ namespace snmalloc
       message<1024>("small_alloc_slow rsize={} slab size={}", rsize, slab_size);
 #endif
 
+      // Calculate the extra bytes required to store the client meta-data.
+      size_t extra_bytes = BackendSlabMetadata::get_extra_bytes(sizeclass);
```

Comment:
Hm.  The reason we're having to do this math here in the frontend is because `Config::Backend::alloc_chunk` doesn't know (and isn't told) the sizeclass?  (The `dealloc_chunk` case above could presumably read out the sizeclass from the pagemap if we didn't want to tell it... but we could also probably just tell it... but in any case, just reading from the pagemap doesn't work for the allocate half anyway.)

Should we be passing in the sizeclass and having the backend do the `get_extra_bytes`?  (And maybe also `public_state()` and having the backend do the `from_small_class` too?)

Reply:
That makes sense I will move it.  What I did was for expediency in getting something working, and I should have revisited this part.